src/acceptance-test/java/org/zalando/nakadi/webservice/timelines/TimelineSyncAT.java

```diff
@@ -0,0 +1,201 @@
+                }
+            }
+        }, "AT_delayed_runs");
+        delayedRunsExecutor.setDaemon(true);
+        delayedRunsExecutor.start();
+    }
+
+    @AfterClass
+    public static void terminate() {
+        delayedRunsExecutor.interrupt();
+        delayedRunsExecutor = null;
+        CURATOR.close();
+    }
+
+    @Test
+    public void testNodeInformationWrittenOnStart() throws Exception {
+        final TimelineSync sync = createTimeline();
+        final String currentVersion = new String(CURATOR.getData().forPath("/nakadi/timelines/version"));
+        Assert.assertEquals(currentVersion,
+                new String(CURATOR.getData().forPath("/nakadi/timelines/nodes/" + sync.getNodeId())));
+    }
+
+    @Test
+    public void testTimelineUpdateWaitsForActivePublish() throws InterruptedException, IOException, TimeoutException {
+        final TimelineSync t1 = createTimeline();
+        final TimelineSync t2 = createTimeline();
+        final String eventType = UUID.randomUUID().toString();
+        final AtomicBoolean updated = new AtomicBoolean(false);
+
+        final Closeable publishing = t1.workWithEventType(eventType, TimeUnit.SECONDS.toMillis(1));
+        try {
+            new Thread(() -> {
+                try {
+                    t2.startTimelineUpdate(eventType, TimeUnit.SECONDS.toMillis(30));
+                    updated.set(true);
+                } catch (final InterruptedException ex) {
+                    throw new RuntimeException(ex);
+                }
+            }, "timeline_update").start();
+            // Wait a little bit for thread to start timeline update
+            Thread.sleep(TimeUnit.SECONDS.toMillis(3));
+            Assert.assertEquals(false, updated.get());
+        } finally {
+            publishing.close();
+        }
+        int attemptsLeft = 5;
+        while (attemptsLeft-- > 0) {
+            // Wait a little bit for lock release
+            Thread.sleep(300);
+            if (updated.get()) {
+                break;
+            }
+        }
+        Assert.assertEquals(true, updated.get());
+    }
+
+    @Test
+    public void testPublishPauseOnTimelineUpdate() throws InterruptedException, IOException {
+        final TimelineSync t1 = createTimeline();
+        final TimelineSync t2 = createTimeline();
+        final String eventType = UUID.randomUUID().toString();
+        // Lock publishing
+        t1.startTimelineUpdate(eventType, TimeUnit.SECONDS.toMillis(30));
+        final AtomicBoolean lockTaken = new AtomicBoolean(false);
+        new Thread(() -> {
+            try (Closeable cl = t2.workWithEventType(eventType, TimeUnit.SECONDS.toMillis(5))) {
+                lockTaken.set(true);
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+        }, "publisher").start();
+
+        Thread.sleep(TimeUnit.SECONDS.toMillis(2)); // Wait for thread to start.
+        Assert.assertEquals(false, lockTaken.get());
```

Comment:
At this position it is not possible. I check, that after 2 seconds publishing is still not started. I don't have to wait for anything else.